pinot-common/src/main/java/com/linkedin/pinot/common/metrics/AbstractMetrics.java

```diff
@@ -267,6 +267,35 @@ public Long call() throws Exception {
   }
 
   /**
+   * Sets the value of a global  gauge.
+   *
+   * @param gauge The gauge to use
+   * @param value The value to set the gauge to
+   */
+  public void setValueOfGlobalGauge(final G gauge, final long value) {
```

Comment:
Since we didn't have global gauges before, can you add the proper initialization code in `AbstractMetrics.initializeGlobalMeters`. What that code does is that it initializes all the metrics to zero at startup so that the metrics get created. Might need to rename that method to `initializeGlobalMetrics`, since it'll do both gauges and meters now.